lib/js/components/TextGroup/TextGroup.vue

```diff
@@ -0,0 +1,296 @@
+<template>
+	<div
+		class="ds-textGroup"
+		:class="{
+			'-ds-small': size === TEXT_GROUP_SIZES.SMALL,
+
+			'-ds-hovered': state === TEXT_GROUP_STATES.HOVERED,
+			'-ds-loading': isLoading,
+			'-ds-disabled': state === TEXT_GROUP_STATES.DISABLED,
+
+			'-ds-interactive': isInteractive,
+			'-ds-selected': isSelected,
+
+			[loadingSizeClassName]: isLoading,
+
+			[colorClassName]: true,
+		}"
+	>
+		<div
+			v-if="eyebrowText !== null"
+			class="ds-textGroup__eyebrow"
+			:class="{
+				'-ds-ellipsis': eyebrowTextEllipsis,
+				'-ds-uppercase': isEyebrowTextUppercase,
+			}"
+		>
+			<ds-skeleton v-if="isLoading" width="50%" height="100%" />
+			<span v-else>{{ eyebrowText }}</span>
+		</div>
+		<div
+			v-if="mainText !== null"
+			class="ds-textGroup__main"
+			:class="{
+				'-ds-ellipsis': mainTextEllipsis,
+			}"
+		>
+			<ds-skeleton v-if="isLoading" width="100%" height="100%" />
+			<span v-else>{{ mainText }}</span>
+		</div>
+		<div
+			v-if="supportingText !== null"
+			class="ds-textGroup__supporting"
+			:class="{
+				'-ds-ellipsis': supportingTextEllipsis,
+			}"
+		>
+			<ds-skeleton v-if="isLoading" width="100%" height="100%" />
+			<span v-else v-html="supportingText" />
+		</div>
+	</div>
+</template>
+
+<style lang="scss">
+@import '../../../styles/settings/typography/tokens';
+@import '../../../styles/settings/colors/tokens';
+@import '../../../styles/settings/spacings';
+
+.ds-textGroup {
+	$self: &;
+
+	display: flex;
+	flex-direction: column;
+	width: 100%;
+
+	&__eyebrow {
+		@include info-s-default-bold;
+
+		color: $color-neutral-text-weak;
```

Comment:
Optional: we could already prepare a map of color for `color` prop, so we don't need to analyze the component again. I suggest similar approach to: https://github.com/bethinkpl/design-system/blob/48818e1ab8537c238ee314967e38fd246f41d3af/lib/js/components/BadgeScore/BadgeScore.vue#L31